### PR TITLE
Add UPGRADING.md

### LINKED ISSUE #70: Add UPGRADING.md

### PR DESCRIPTION
**Pull request by vweevers**
Closes #70.

Many things mentioned here can be moved to the readme later (#123).

### CODE REVIEW

File: UPGRADING.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+-   ArrayBuffer or a view thereof (typed arrays);
+-   Map, Set, Blob, File, FileList, ImageData (limited support).
+
+In addition `level-js` now stores `Buffer` values without transformation. This works in all target environments because modern `Buffer` is a subclass of `Uint8Array`, meaning such values can be passed to `IndexedDB` as-is.
+
+If the environment does not support a type, it will throw an error which `level-js` catches and passes to the callbacks of `put` or `batch`. For example, IE does not support typed array values. At the time of writing, Chrome is the only browser that supports all types listed above.
+
+Due to the special meaning that `null` and `undefined` have in `abstract-leveldown` iterators and Node.js streams, values of this type are converted to empty strings prior to storage.
+
+:point_up: The structured clone algorithm can be slower than `JSON.stringify`, so don't to quick to eschew `encoding-down`.
```
Comment (vweevers) [inline UPGRADING.md]: to => too

Comment (ralphtheninja) [reply UPGRADING.md]: When wrapping with level-browserify, maybe we should provide an option to use encoding-down or not?

Comment (vweevers) [reply UPGRADING.md]: IMO `levelup(leveljs())` is just fine, or even better for composition than options.

**Comment by vweevers** [comment]
@ralphtheninja if you want I can move some stuff to the readme as part of this PR, closing #123. Then you can review everything in one go and we can release 3.0.0 (skipping rc2).

File: UPGRADING.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+### Type support
+
+All value types of the [structured clone algorithm][structured-clone-algorithm] and all key types of IndexedDB Second Edition are now supported. This means you can store almost any JavaScript type without the need for [`encoding-down`][encoding-down]. In addition, you can use [`Buffer`][buffer] for both keys and values. For details and caveats please see the [readme][readme].
+
+### No backpressure
+
+In `level-js`, iterators are powered by IndexedDB cursors. To fulfill [`abstract-leveldown`][abstract-leveldown] snapshot guarantees (reads not being affected by simultaneous writes) cursors are started immediately and continuously read from, filling an in-memory cache.
+
+Though `level-js` now passes the full [`abstract-leveldown`][abstract-leveldown] test suite, fulfulling the snapshot guarantee means a loss of backpressure. Memory consumption might increase if an iterator is not consumed fast enough. A future release will have an option to favor backpressure over snapshot guarantees.
```
Comment (ralphtheninja) [inline UPGRADING.md]: `fulfilling`

**Pull request merged**